components/raftstore/src/store/fsm/peer.rs

```diff
@@ -1260,9 +1260,58 @@ where
             SignificantMsg::RaftlogFetched { context, res } => {
                 self.on_raft_log_fetched(context, res);
             }
+            SignificantMsg::EnterForceLeaderState => self.on_enter_force_leader(),
+            SignificantMsg::ExitForceLeaderState => self.on_exit_force_leader(),
         }
     }
 
+    fn on_enter_force_leader(&mut self) {
+        info!(
+            "enter force leader state";
+            "region_id" => self.fsm.region_id(),
+            "peer_id" => self.fsm.peer_id(),
+        );
+        self.fsm.peer.force_leader = true;
+
+        if !self.fsm.peer.is_leader() {
+            // become candidate first to increase term
+            self.fsm.peer.raft_group.raft.become_candidate();
+            // trigger candidate twice to increase term by 2, to avoid there is
+            // a existing leader by accident.
+            self.fsm.peer.raft_group.raft.become_candidate();
+            self.fsm.peer.raft_group.raft.become_leader();
+        }
+        assert!(self.fsm.peer.is_leader());
+        self.fsm.peer.raft_group.raft.set_check_quorum(false);
+
+        // forward commit index
+        self.fsm.peer.raft_group.raft.raft_log.committed =
```

Comment:
What if `committed` is larger than `persisted`?

Reply:
If there is a restart, how are some of the committed logs recovered as they aren't persisted?